comicapi/genericmetadata.py

```diff
@@ -202,24 +212,122 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            getattr(self, cur).difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_dedupe_cur(self, new: list[Any] | set[Any], cur: str) -> None:
+        if isinstance(new, list):
+            remove_ids = set()
+            for a in getattr(self, cur):
+                for i, b in enumerate(new):
+                    if a.casefold() == b.casefold():
+                        remove_ids.add(i)
+                        break
+            for r_id in remove_ids:
+                new.pop(r_id)
+            getattr(self, cur).extend(new)
+
+        if isinstance(new, set):
+            remove_set = set()
+            for a in getattr(self, cur):
+                for b in new:
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            new.difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_original(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, str) and len(new) == 0:
+                if isinstance(getattr(self, cur), (list, set)):
+                    getattr(self, cur).clear()
+                else:
+                    setattr(self, cur, None)
+            elif isinstance(new, (list, set)) and len(new) == 0:
+                pass
+            else:
+                setattr(self, cur, new)
+
+    def assign_overwrite(self, cur: str, new: Any) -> None:
+        if isinstance(new, (str, list, set)) and len(new) == 0:
+            if isinstance(getattr(self, cur), (list, set)):
+                getattr(self, cur).clear()
+            else:
+                setattr(self, cur, None)
+        else:
+            setattr(self, cur, new)
+
+    def assign_overwrite_no_clear(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, (str, list, set)) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (bool, int, float)):
+                setattr(self, cur, new)
+
+    def assign_add_missing(self, cur: str, new: Any) -> None:
+        if new is not None and not getattr(self, cur):
+            setattr(self, cur, new)
+
+    def assign_combine_new(self, cur: str, new: Any) -> None:
+        # Combine lists and sets and prefer 'new' for strings, floats and ints
+        if new is not None:
+            if isinstance(new, (list, set)) and len(new) > 0 and isinstance(getattr(self, cur), (list, set)):
+                # Check for dupes with casefold to replace accented chars
+                self.assign_dedupe_new(new, cur)
+            elif isinstance(new, str) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (int, float)) and new > 0:
+                setattr(self, cur, new)
```

Comment:
Why is 0 ignored? `None` is used for empty. This makes it impossible to update an issue number 0 such as https://www.dc.com/comics/batman-2011/batman-0

Reply:
That makes issue number a special case? Is there anything else that could be considered that you'd want to overwrite a value `> 0` with `0`? 

`None` is for empty but a talker might return a 0 for `null/None`, is the talker expected to decide if `0` should be `None`?

Reply:
Yes, if the value is different from the default (None for most types) it's assumed that it was set on purpose

Reply:
The current condition also disallows setting any negative numbers. The whole reason int values default to None is so that all integer values can be used